alluxio-python-library/alluxio/alluxio_file_system.py

```diff
@@ -165,13 +165,38 @@ def read(self, file_path):
         worker_host = self._get_preferred_worker_host(file_path)
         path_id = self._get_path_hash(file_path)
         try:
-            return b"".join(self._page_generator(worker_host, path_id))
+            return b"".join(self._all_page_generator(worker_host, path_id))
         except Exception as e:
             raise type(e)(
                 f"Error when reading file {file_path}: error {e}"
             ) from e
 
-    def _page_generator(self, worker_host, path_id):
+    def read_range(self, file_path, offset, length):
```

Comment:
As the function will be used by users, we probably wouldn't expect all inputs to be valid. Could you add some input validation? (eg. could the offset be a negative number?)